csrc/scheduler/utils.cpp

```diff
@@ -2263,31 +2264,64 @@ std::vector<int64_t> domainReorderAsLogicalMap(TensorView* tv) {
   return *permutation;
 }
 
-std::unordered_map<int64_t, int64_t> maybeReorderAsAllocationMap(
+std::unordered_map<int64_t, int64_t> maybeReorderLogicalAsAllocationMap(
     TensorView* tv) {
-  std::unordered_map<int64_t, int64_t> ret;
+  std::unordered_map<int64_t, int64_t> reorder_map;
   if (!tv->hasAllocation()) {
-    return ret;
+    return reorder_map;
   }
-  const auto& alloc_dom = tv->getAllocationDomain();
-  const auto& loop_dom = tv->getLoopDomain();
-  if (alloc_dom == loop_dom) {
-    return ret;
-  }
-  if (!std::is_permutation(
-          alloc_dom.begin(), alloc_dom.end(), loop_dom.begin())) {
-    return ret;
+  const auto& logical_domain = tv->getLogicalDomain();
+  std::optional<Layout> layout = canonicalizeLayout(tv);
+  NVF_ERROR(
+      layout.has_value(),
+      "Failed to canonicalize layout of ",
+      tv->domain()->toString(0, false));
+  const auto& alloc_domain = layout->allocation_domain();
+  if (alloc_domain == logical_domain) {
+    return reorder_map;
   }
-  std::unordered_map<IterDomain*, int64_t> alloc_index;
-  std::unordered_map<IterDomain*, int64_t> rfactor_index;
-  for (auto i : arange((int64_t)alloc_dom.size())) {
-    alloc_index[alloc_dom[i]] = i;
-    rfactor_index[loop_dom[i]] = i;
+  std::optional<std::vector<int64_t>> permutation =
+      ir_utils::computePermutation(alloc_domain, logical_domain);
+  int64_t idx = 0;
+  std::transform(
```

Comment:
This can be simplified by directly using the permutation. Since that requires changing the heuristic entry, I plan to do it separately.